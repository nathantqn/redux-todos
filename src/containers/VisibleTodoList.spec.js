// @flow

import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import VisibleTodoList from './VisibleTodoList';
import { toggleTodo, setVisibilityFilter } from '../actions';

const setup = (setupProps = {}) => {
  const store = configureStore()({
    todos: [
      {
        text: 'Leo',
        completed: false,
        id: 0
      },
      {
        text: 'Nhat',
        completed: true,
        id: 1
      }
    ],
    visibilityFilter: 'SHOW_ALL'
  });
  const wrapper = shallow(<VisibleTodoList store={store} />);

  return {
    store,
    wrapper
  };
};

describe('VisibleTodoList', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('shows all todos when SHOW_ALL filter is active', () => {
    const { store, wrapper } = setup();
    const deeperWrapper = wrapper.shallow();
    store.dispatch(setVisibilityFilter('SHOW_ALL'));
    
    expect(store.getActions()).toEqual([setVisibilityFilter('SHOW_ALL')]);

  });

  it('shows active todos when SHOW_ACTIVE filter is active', () => {
    const { store, wrapper } = setup();
    const deeperWrapper = wrapper.shallow();    
    store.dispatch(setVisibilityFilter('SHOW_ACTIVE'));
    
    expect(store.getActions()).toEqual([setVisibilityFilter('SHOW_ACTIVE')]);
  });

  it('shows completed todos when SHOW_COMPLETED filter is active', () => {
    const { store, wrapper } = setup();
    const deeperWrapper = wrapper.shallow();
    store.dispatch(setVisibilityFilter('SHOW_COMPLETED'));

    expect(store.getActions()).toEqual([setVisibilityFilter('SHOW_COMPLETED')]);
  });

  it('toggles todos when a todo is clicked', () => {
    const { store, wrapper } = setup();
   
    wrapper.shallow().find('Todo').last().simulate('click');
    expect(store.getActions()).toEqual([toggleTodo(1)]);
  });
});