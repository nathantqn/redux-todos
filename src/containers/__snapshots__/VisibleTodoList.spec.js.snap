// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibleTodoList renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(TodoList)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "todos": Array [
        Object {
          "completed": false,
          "id": 0,
          "text": "Leo",
        },
        Object {
          "completed": true,
          "id": 1,
          "text": "Nhat",
        },
      ],
      "toggleTodo": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "todos": Array [
          Object {
            "completed": false,
            "id": 0,
            "text": "Leo",
          },
          Object {
            "completed": true,
            "id": 1,
            "text": "Nhat",
          },
        ],
        "toggleTodo": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
